### PR TITLE
Add Trino 419 release notes

### LINKED ISSUE #17546: Release notes for 419

### PR DESCRIPTION
**Pull request by colebow**
## Description

Assemble the release notes for the upcoming Trino 419 release. Fixes #17546 

## Release notes

(x) This is not user-visible or docs only and no release notes are required.

## Verification for each pull request

Format: maintainer, PR/issue number, ✅ / ❌ rn ✅ / ❌ docs
✅ rn - release note added and verified, or assessed to be not necessary, set to ❌ rn before completion
✅ docs - need for docs assessed and merged, or assessed to be not necessary, set to ❌ docs before completion

## 18 May 2023

* #17524 ✅ rn ✅ docs
* #17530 ✅ rn ✅ docs
* #16745 ✅ rn ✅ docs
* #17557 ✅ rn ✅ docs

## 19 May 2023

* #17547 ✅ rn ✅ docs
* #17567 ✅ rn ✅ docs
* #17462 ✅ rn ✅ docs
* #17467 ✅ rn ✅ docs
* #17570 ✅ rn ✅ docs
* #17552 ✅ rn ✅ docs
* #17212 ✅ rn ✅ docs
* #17574 ✅ rn ✅ docs
* #17569 ✅ rn ✅ docs

## 21 May 2023

* #17539 ✅ rn ✅ docs
* #17581 ✅ rn ✅ docs
* #17548 ✅ rn ✅ docs
* #17584 ✅ rn ✅ docs
* #17578 ✅ rn ✅ docs

## 22 May 2023

* #17590 ✅ rn ✅ docs
* #17571 ✅ rn ✅ docs
* #17171 ✅ rn ✅ docs
* #17591 ✅ rn ✅ docs
* #17517 ✅ rn ✅ docs
* #16024 ✅ rn ✅ docs

## 23 May 2023

* #17603 ✅ rn ✅ docs
* #17580 ✅ rn ✅ docs
* #17550 ✅ rn ✅ docs
* #17196 ✅ rn ✅ docs
* #17609 ✅ rn ✅ docs
* #17551 ✅ rn ✅ docs
* #17585 ✅ rn ✅ docs
* #17601 ✅ rn ✅ docs
* #17587 ✅ rn ✅ docs
* #17599 ✅ rn ✅ docs

## 24 May 2023

* #17615 ✅ rn ✅ docs
* #17612 ✅ rn ✅ docs
* #17014 ✅ rn ✅ docs
* #17469 ✅ rn ✅ docs
* #17460 ✅ rn ✅ docs
* #17458 ✅ rn ✅ docs
* #17607 ✅ rn ✅ docs
* #17268 ✅ rn ✅ docs
* #17252 ✅ rn ✅ docs
* #17625 ✅ rn ✅ docs
* #17630 ✅ rn ✅ docs
* #17067 ✅ rn ❌ docs

## 25 May 2023

* #17556 ✅ rn ✅ docs
* #17638 ✅ rn ✅ docs
* #17622 ✅ rn ✅ docs
* #17629 ✅ rn ✅ docs
* #17617 ✅ rn ✅ docs
* #17645 ✅ rn ✅ docs
* #17632 ✅ rn ✅ docs

## 26 May 2023

* #17560 ✅ rn ✅ docs
* #17650 ✅ rn ✅ docs
* #17596 ✅ rn ✅ docs
* #17575 ✅ rn ✅ docs
* #17360 ✅ rn ✅ docs
* #17652 ✅ rn ✅ docs

## 27 May 2023

* #17633 ✅ rn ✅ docs

## 28 May 2023

* #11701 ✅ rn ✅ docs
* #17414 ✅ rn ✅ docs

## 29 May 2023

* #17653 ✅ rn ✅ docs
* #17236 ✅ rn ✅ docs
* #17665 ✅ rn ✅ docs
* #17678 ✅ rn ✅ docs
* #14246 ✅ rn ✅ docs
* #17655 ✅ rn ✅ docs
* #17691 ✅ rn ✅ docs

## 30 May 2023

* #17685 ✅ rn ✅ docs
* #17680 ✅ rn ✅ docs
* #17699 ✅ rn ✅ docs
* #17700 ✅ rn ✅ docs
* #17695 ✅ rn ✅ docs
* #17706 ✅ rn ✅ docs
* #17698 ✅ rn ✅ docs

## 31 May 2023

* #17701 ✅ rn ✅ docs
* #17554 ✅ rn ✅ docs
* #17711 ✅ rn ✅ docs
* #17704 ✅ rn ✅ docs

## 1 June 2023

* #17723 ✅ rn ✅ docs
* #17724 ✅ rn ✅ docs
* #17726 ✅ rn ✅ docs
* #17715 ✅ rn ✅ docs
* #17728 ✅ rn ✅ docs
* #17718 ✅ rn ✅ docs
* #17498 ❌ rn ❌ docs
* #16064 ✅ rn ✅ docs

## 2 June 2023

* #17722 ✅ rn ✅ docs
* #16879 ✅ rn ✅ docs
* #17624 ✅ rn ✅ docs
* #17732 ✅ rn ✅ docs
* #17392 ✅ rn ✅ docs
* #17730 ✅ rn ✅ docs
* #17507 ✅ rn ✅ docs
* #17741 ✅ rn ✅ docs
* #17739 ✅ rn ✅ docs

## 4 June 2023

* #17737 ✅ rn ✅ docs

## 5 June 2023

* #17736 ✅ rn ✅ docs
* #17742 ✅ rn ✅ docs

### CODE REVIEW

File: docs/src/main/sphinx/release/release-419.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+* Fix JSON serialization failure when using string columns with values larger
+  than 5 MB. ({issue}`17557`)
+
+## Delta Lake connector
+
+* Improve performance of reading from Parquet files. ({issue}`17612`)
+* Improve display of table handles in the output of `EXPLAIN` queries. ({issue}`17590`)
+* Fix query failure when writing to a file location with a trailing `/` in its
+  name. ({issue}`17552`)
+* Fix sporadic query failure. ({issue}`17581`)
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-419.md:16]: Not sure that is any useful

File: docs/src/main/sphinx/release/release-419.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  `map` and `row` types. Previously, this returned incorrect results after the
+  migration. ({issue}`17587`)
+* Improve performance of reading ORC files with Bloom filter indexes. ({issue}`17530`)
+* Improve performance of reading from Parquet files. ({issue}`17612`)
+* Improve performance of reading table statistics. ({issue}`16745`)
+* Fix sporadic query failure. ({issue}`17581`)
+
+## Kudu connector
+
+* Fix sporadic query failure. ({issue}`17462`)
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-419.md:41]: Different PR same message?

**Review by mosabua — CHANGES_REQUESTED** [review]

File: docs/src/main/sphinx/release/release-419.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
+# Release 419 (24 May 2023)
+
+## General
+
+* Add the {func}`array_histogram` function to find the number of occurrences of
+  the unique elements in an array. ({issue}`14725 `)
+* Fix JSON serialization failure when using string columns with values larger
```
Comment (martint) [inline docs/src/main/sphinx/release/release-419.md:7]: How does it manifest? Is it a client-coordinator failure? A worker-worker communication failure?

I would just say "Fix failure when processing string columns with values larger than 5 MB", or something to that effect.

cc @hashhar @wendigo

File: docs/src/main/sphinx/release/release-419.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+  than 5 MB. ({issue}`17557`)
+
+## Delta Lake connector
+
+* Improve performance of reading from Parquet files. ({issue}`17612`)
+* Improve display of table handles in the output of `EXPLAIN` queries. ({issue}`17590`)
```
Comment (martint) [inline docs/src/main/sphinx/release/release-419.md:13]: "Table handle" is an internal concept. How is the display improved? Describe in terms of user-visible impact. cc @losipiuk

File: docs/src/main/sphinx/release/release-419.md (hunk 3 above)
Comment (wendigo) [reply docs/src/main/sphinx/release/release-419.md:7]: Actually it can be both @martint

**Review by mosabua — APPROVED** [review]
Looks good now.

File: docs/src/main/sphinx/release/release-419.md [hunk 5]
```diff
@@ -0,0 +1,33 @@
+# Release 419 (24 May 2023)
+
+## General
+
+* Add the {func}`array_histogram` function to find the number of occurrences of
+  the unique elements in an array. ({issue}`14725 `)
+* Fix failure when processing string columns with values larger than 5 MB. ({issue}`17557`)
```
Comment (hashhar) [inline docs/src/main/sphinx/release/release-419.md:7]: this is inaccurate and downplays the cases where it can happen. It can happen anytime the server sends the client a response with a string value larger than 5MB. The string value doesn't have to be a value from the table, it can even be things like query stats that we send.

IMO the wording here should be to avoid all Trino JDBC and Trino CLI >= 415 and < 419.

Comment (hashhar) [reply docs/src/main/sphinx/release/release-419.md:7]: It's just easiest to reproduce with a table having large string values but that's only one of the possible failures.

Comment (wendigo) [reply docs/src/main/sphinx/release/release-419.md:7]: Since it applies to deserialization path I'd phrase it like: `fix query failures when server JSON responses exceed 5MB length limit for string values`

File: docs/src/main/sphinx/release/release-419.md [hunk 6]
```diff
@@ -1,15 +1,29 @@
...
+  configuration property. ({issue}`17334`)
+
+## BigQuery connector
+
+* Add support for proxying BigQuery APIs via an HTTP(S) proxy. ({issue}`17508`)
+* Improve performance of retrieving metadata from BigQuery. ({issue}`16064`)
+
 ## Delta Lake connector
 
+* Support adding a new column with the `id` and `name` column mapping modes. ({issue}`17236`)
```
Comment (martint) [inline docs/src/main/sphinx/release/release-419.md:23]: Are those two mutually exclusive modes?

Maybe rephrase as "Support the id and name mapping modes when adding new columns".

File: docs/src/main/sphinx/release/release-419.md [hunk 7]
```diff
@@ -19,16 +33,33 @@
 
 * Improve performance of reading ORC files with Bloom filter indexes. ({issue}`17530`)
 * Improve performance of reading Parquet files. ({issue}`17612`)
+* Improve optimized Parquet writer performance for
+  [structural data types](structural-data-types). ({issue}`17665`)
+* Fix query failure when reading ORC files with a shorthand timezone id in the
```
Comment (martint) [inline docs/src/main/sphinx/release/release-419.md:38]: This is confusing. It adds new behavior to support legacy shorthand timezones, but it's listed as "Fix query failure". I would rephrase as "Add support for reading ORC files with shorthand timezone ids in the stripe footer metadata".

File: docs/src/main/sphinx/release/release-419.md [hunk 8]
```diff
@@ -19,16 +33,33 @@
...
   migration. ({issue}`17587`)
 * Improve performance of reading ORC files with Bloom filter indexes. ({issue}`17530`)
 * Improve performance of reading Parquet files. ({issue}`17612`)
+* Improve performance when writing Parquet files with
+  [structural data types](structural-data-types). ({issue}`17665`)
 * Improve performance of reading table statistics. ({issue}`16745`)
+
+## SPI
+
+* Fix incorrect behavior when using the `DEFAULT_ON_NULL` return convention. ({issue}`17706`)
```
Comment (martint) [inline docs/src/main/sphinx/release/release-419.md:64]: This is a fix to a change that hasn't been released (https://github.com/trinodb/trino/pull/17212), so it doesn't need a release note item.

File: docs/src/main/sphinx/release/release-419.md [hunk 9]
```diff
@@ -0,0 +1,64 @@
+# Release 419 (5 June 2023)
```
Comment (ebyhr) [inline docs/src/main/sphinx/release/release-419.md:1]: We've used `Jun` instead of `June` in the past release notes.

Comment (mosabua) [reply docs/src/main/sphinx/release/release-419.md:1]: @ryandoowan can you check and maybe send a small PR to fix this .. cc @colebow
